proto/sources.proto

```diff
@@ -270,3 +271,14 @@ message SlackRealtime {
     credentials.SlackTokens tokens = 1;
   }
 }
+
+message Sharepoint {
+  string endpoint = 1 [(validate.rules).string.uri_ref = true];
+  oneof credential {
+    string token = 2;
+    credentials.ClientCredentials authenticated = 3;
+    credentials.Oauth2 oauth = 4;
+  }
+  string site_url = 5;
```

Comment:
Not sure if we need this but we can leave it for now. We should enumerate if possible